MushroomObserver/mushroom-observer
# Non-numeric ICN ID

Fixes a bug which set ICN ID to zero when user entered non-numeric ID (e.g., "MB28460"). [1]

### Manual Test Script
- Edit any fungal name, entering a non-numeric ID. 
Expected Result:
- Form is reloaded
- Flash Error displayed:
>Unable to save changes.
Name ICN Identifier is not a number. 

Delivers https://www.pivotaltracker.com/story/show/175730994

---

[1] The bug is actually more complex.
The first time any user saves a non-numeric `icn_id`, MO will:
- flash success
- save the ICN id as 0 in the db
- show_name page will say "ICN Identifier: _missing_"

If someone saves a non-numeric `icn_id` for a second Name:
- MO tries to save `icn_id` as 0, but fails 
- flashes error "An ICN identifier number may be used by only one Name. ‘0’ is already used by ‘\<first Name>’."
